cfgov/unprocessed/apps/youth-employment-success/js/util.js

```diff
@@ -0,0 +1,110 @@
+let UNDEFINED;
+
+const REDUCER_RETURN_ERROR = 'Reducer must return a state object';
+const INVALID_ARG_ERROR = 'reducers argument must be an object, where each value is a reducer function';
+
+/**
+ *
+ * @param {object} state Composed of key / value pairs where the key is the name of
+ *  a slice of app state, and the value is the values for that slice of state
+ * @param {string} key name of the state slice
+ * @returns {function|undefined} Return the reducer function or undefined
+ */
+function checkForPreviousState( state = {}, key ) {
+  if ( state[key] ) {
+    return state[key];
+  }
+
+  return UNDEFINED;
+}
+
+/**
+ * Throw an error if the reducer processing the current action doesn't return
+ * a value
+ * @param {object} nextState The state object just returned by a reducer
+ */
+function assertNextState( nextState ) {
+  if ( typeof nextState === 'undefined' ) {
+    throw new Error( REDUCER_RETURN_ERROR );
+  }
+}
+
+const processStateFromReducers = reducers => (
+  state = UNDEFINED,
+  action = { type: null }
+) => {
+  const reducersKeys = Object.keys( reducers );
+  const finalState = {};
+  let hasChanged = false;
+
+  for ( let i = 0; i < reducersKeys.length; i++ ) {
+    const key = reducersKeys[i];
+    const reducer = reducers[key];
+
+    /**
+     * If we didnt already get a composite state object, e.g. if this is
+     * the first time the reducers are running, default to undefined
+     * and force the reducer to use it's default state.
+     */
+    const previousState = checkForPreviousState( state, key );
+    const argumentsToApply = [ previousState, action ];
+    const nextState = reducer.apply( reducer, argumentsToApply );
+
+    assertNextState( nextState );
+
+    finalState[key] = nextState;
+    hasChanged = hasChanged || nextState !== previousState;
+  }
+
+  return hasChanged ? finalState : state;
+};
+
+export const combineReducers = reducers => {
```

Comment:
Would it make sense to define `const combineReducers = …` here and then at the end of the file have `export combineReducers;`, so that all exports are together?

Reply:
yes, I will normalize that for sure in my next PR!